goodconf/__init__.py

```diff
@@ -112,6 +102,9 @@ def initial_for_field(name: str, field_info: FieldInfo) -> Any:
     return ""
 
 
+CONFIG_CONTEXT = ContextVar("_goodconf_config")
```

Comment:
This might need some consideration whether this is threadsafe or not. I think it might be since a single thread can only call `load` on one settings class concurrently.